src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerListForHearing.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmcts.divorce.caseworker.event;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.ccd.sdk.type.ListValue;
+import uk.gov.hmcts.divorce.common.ccd.PageBuilder;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.Hearing;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+
+import java.util.UUID;
+
+import static java.util.Collections.singletonList;
+import static java.util.Objects.isNull;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.PendingHearingDate;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.PendingHearingOutcome;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.JUDGE;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE_DELETE;
+
+@Component
+@Slf4j
+public class CaseworkerListForHearing implements CCDConfig<CaseData, State, UserRole> {
+    public static final String CASEWORKER_LIST_FOR_HEARING = "caseworker-list-for-hearing";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_LIST_FOR_HEARING)
+            .forStateTransition(
+                PendingHearingDate, PendingHearingOutcome
+            )
+            .name("List for hearing")
+            .description("List for hearing")
+            .showEventNotes()
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .showSummary()
+            .grant(CREATE_READ_UPDATE, CASE_WORKER)
+            .grant(CREATE_READ_UPDATE_DELETE, SUPER_USER)
```

Comment:
If they are going to be able to delete, should we then be showing all the hearings on this page for deletion?
Otherwise they won't be able to edit them as they will be in the Hearings data for existing, and only shows empty hearing data object I think?